batchflow/models/tf/layers/core.py

```diff
@@ -2,78 +2,101 @@
 import numpy as np
 import tensorflow as tf
 
+from .utils import add_as_function
 
-def flatten2d(inputs, name=None):
+
+
+@add_as_function
+class Flatten2D:
     """ Flatten tensor to two dimensions (batch_size, item_vector_size) """
-    x = tf.convert_to_tensor(inputs)
-    dims = tf.reduce_prod(tf.shape(x)[1:])
-    x = tf.reshape(x, [-1, dims], name=name)
-    return x
+    def __init__(self, *args, **kwargs):
+        self.args, self.kwargs = args, kwargs
 
+    def __call__(self, inputs):
+        x = tf.convert_to_tensor(inputs)
+        dims = tf.reduce_prod(tf.shape(x)[1:])
+        x = tf.reshape(x, [-1, dims], **self.kwargs)
+        return x
 
-def flatten(inputs, name=None):
-    """ Flatten tensor to two dimensions (batch_size, item_vector_size) using inferred shape and numpy """
-    x = tf.convert_to_tensor(inputs)
-    shape = x.get_shape().as_list()
-    dim = np.prod(shape[1:])
-    x = tf.reshape(x, [-1, dim], name=name)
-    return x
 
 
-def alpha_dropout(inputs, rate=0.5, noise_shape=None, seed=None, training=False, name=None):
```

Comment:
Did you remove this functionality completely?

Reply:
It is present in `Keras`. Restored and assigned to `D` letter in `layout`-convention